cypress/integration/login.spec.js

```diff
@@ -0,0 +1,44 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Login and OAuth authorization', () => {
+  it('Displays an error on invalid login credentials', () => {
+    cy.visit('/console')
+
+    cy.grab('login__form__user_name')
+      .type('adminwrong')
+      .should('have.value', 'adminwrong')
+    cy.grab('login__form__password')
+      .type('adminwrong')
+      .should('have.value', 'adminwrong')
+      .type('{enter}')
+
+    cy.location('pathname').should('include', '/oauth')
+
+    cy.findAllByText('incorrect password or user ID').should('exist')
```

Comment:
> I dont think we should check for the message as it is returned by the backend. All we care about here is that the path has no `/console` and the error notification is present on the screen.

I disagree. The goal is to make the tests resemble the actual user experience as closely as possible. In this test case, a user would expect to be shown a notification that informs about the nature of the problem that lead to the submit not being successful. If we only check for the error notification, the test would also pass when there is a notification displaying irrelevant or no information. Generally speaking, the more we match realistic user expectations in our tests, the more confidence we can derive from them.

Also note that since this is an end-to-end test, it is legitimate to make assertions based on backend behavior.

Do you agree?